File: app/src/main/java/com/avidprogrammers/insurancepremiumcalculator/NotificationActivity.java
package com.avidprogrammers.insurancepremiumcalculator;

import android.content.Context;
import android.content.Intent;
import android.content.IntentFilter;
import android.content.SharedPreferences;
import android.net.ConnectivityManager;
import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;
import android.view.MenuItem;

import com.avidprogrammers.app.Config;
import com.avidprogrammers.utils.NotifiationAdapter;
import com.google.android.gms.ads.AdRequest;
import com.google.android.gms.ads.AdView;

import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;


public class NotificationActivity extends AppCompatActivity implements ConnectivityReceiver.ConnectivityReceiverListener {

    private AdView mAdView;
    private RecyclerView recyclerView;
    private RecyclerView.LayoutManager layoutManager;
    private NotifiationAdapter adapter;

    ConnectivityReceiver conn;

    CheckingStatus checkingStatus;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_notification);
        SharedPreferences pref = getApplicationContext().getSharedPreferences(Config.SHARED_PREF, 0);
        SharedPreferences.Editor editor = pref.edit();
        editor.putInt("newNoti", 0);
        editor.commit();
        //setTitle("Notifications");
        checkingStatus=new CheckingStatus();
        conn=new ConnectivityReceiver();
        IntentFilter intentFilter=new IntentFilter();
        intentFilter.addAction(ConnectivityManager.CONNECTIVITY_ACTION);
        registerReceiver(conn, intentFilter);
        checkfunction(NotificationActivity.this);

/*        mAdView = (AdView) findViewById(R.id.adView);
        AdRequest adRequest = new AdRequest.Builder().build();
        mAdView.loadAd(adRequest);*/

        recyclerView = findViewById(R.id.notification_recyclerview);
        layoutManager = new LinearLayoutManager(this);
        adapter = new NotifiationAdapter(this);
        recyclerView.setLayoutManager(layoutManager);
        recyclerView.setAdapter(adapter);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
    }

    @Override
    public void onNetworkConnectionChanged(boolean isConnected) {
        checkingStatus.notification(isConnected,this);

    }

    @Override
    protected void onResume() {
        super.onResume();
        MyApplication.getInstance().setConnectivityListener(this);
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
        unregisterReceiver(conn);
    }

    public void checkfunction(Context context){
        boolean isConnected=ConnectivityReceiver.isConnected();
        //notification(isConnected,lp_taxi_upto18pass.this);
        checkingStatus.notification(isConnected,context);

    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch(item.getItemId())
        {
            case android.R.id.home:
                startActivity(new Intent(NotificationActivity.this,home_activity.class));
                finish();
                break;
        }
        return true;
    }

    @Override
    public void onBackPressed() {
        super.onBackPressed();
        startActivity(new Intent(NotificationActivity.this,home_activity.class));
        finish();
    }
}

File: app/src/main/java/com/avidprogrammers/insurancepremiumcalculator/NotificationActivity.kt
package com.avidprogrammers.insurancepremiumcalculator

import android.content.Context
import android.content.Intent
import android.content.IntentFilter
import android.content.SharedPreferences
import android.net.ConnectivityManager
import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.avidprogrammers.app.Config
import com.avidprogrammers.insurancepremiumcalculator.ConnectivityReceiver.ConnectivityReceiverListener
import com.avidprogrammers.utils.NotifiationAdapter
import com.google.android.gms.ads.AdView

class NotificationActivity : AppCompatActivity(), ConnectivityReceiverListener {
    private val mAdView: AdView? = null
    private var recyclerView: RecyclerView? = null
    private var layoutManager: RecyclerView.LayoutManager? = null
    private var adapter: NotifiationAdapter? = null
    var conn: ConnectivityReceiver? = null
    var checkingStatus: CheckingStatus? = null
    protected override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_notification)
        val pref: SharedPreferences =
            getApplicationContext().getSharedPreferences(Config.SHARED_PREF, 0)
        val editor = pref.edit()
        editor.putInt("newNoti", 0)
        editor.commit()
        //setTitle("Notifications");
        checkingStatus = CheckingStatus()
        conn = ConnectivityReceiver()
        val intentFilter = IntentFilter()
        intentFilter.addAction(ConnectivityManager.CONNECTIVITY_ACTION)
        registerReceiver(conn, intentFilter)
        checkfunction(this@NotificationActivity)

/*        mAdView = (AdView) findViewById(R.id.adView);
        AdRequest adRequest = new AdRequest.Builder().build();
        mAdView.loadAd(adRequest);*/recyclerView =
            findViewById<RecyclerView>(R.id.notification_recyclerview)
        layoutManager = LinearLayoutManager(this)
        adapter = NotifiationAdapter(this)
        recyclerView!!.layoutManager = layoutManager
        recyclerView!!.adapter = adapter
        getSupportActionBar()!!.setDisplayHomeAsUpEnabled(true)
    }

    override fun onNetworkConnectionChanged(isConnected: Boolean) {
        checkingStatus!!.notification(isConnected, this)
    }

    protected override fun onResume() {
        super.onResume()
        MyApplication.Companion.instance!!.setConnectivityListener(this)
    }

    protected override fun onDestroy() {
        super.onDestroy()
        unregisterReceiver(conn)
    }

    fun checkfunction(context: Context?) {
        val isConnected: Boolean = ConnectivityReceiver.Companion.isConnected
        //notification(isConnected,lp_taxi_upto18pass.this);
        checkingStatus!!.notification(isConnected, context!!)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            android.R.id.home -> {
                startActivity(Intent(this@NotificationActivity, home_activity::class.java))
                finish()
            }
        }
        return true
    }

    override fun onBackPressed() {
        super.onBackPressed()
        startActivity(Intent(this@NotificationActivity, home_activity::class.java))
        finish()
    }
}